hydrabus_framework/utils/hb_generic_cmd.py

```diff
@@ -0,0 +1,78 @@
+import time
+import serial
+from hydrabus_framework.utils.logger import Logger
+
+
+__author__ = "Jordan Ovrè <ghecko78@gmail.com>"
+
+
+def hb_reset(serial_instance):
+    """
+    Reset hydrabus to return in console mode
+    :param serial_instance: hydrabus serial instance
+    """
+    serial_instance.write(b'\x00')
+    serial_instance.write(b'\x0F')
+    time.sleep(0.2)
+    # clean serial buffer
+    serial_instance.read(serial_instance.in_waiting)
+
+
+def hb_close(serial_instance):
+    """
+    Close hydrabus serial instance
+    :param serial_instance: hydrabus serial instance
+    """
+    serial_instance.close()
+
+
+def hb_switch_bbio(serial_instance):
+    """
+    Init the hydrabus to switch into BBIO mode
+    :param serial_instance: hydrabus serial instance
+    :return: Bool
+    """
+    for i in range(20):
+        serial_instance.write(b'\x00')
+    if "BBIO1".encode('utf-8') not in serial_instance.read(5):
+        return False
+    return True
+
+
+def hb_wait_ubtn(serial_instance):
+    """
+    Loop until user press hydrabus UBTN
+    :param serial_instance: hydrabus serial instance
+    :return: Nothing
+    """
+    # timeout=1 minutes
+    timeout = time.time() + 60 * 1
+    while True:
+        if serial_instance.read(1) == 'B'.encode('utf-8'):
+            if serial_instance.read(3) == 'BIO'.encode('utf-8'):
+                # carriage return needed to reset interface
+                serial_instance.write(b'\x0D\x0A')
+                time.sleep(0.2)
+                serial_instance.read(serial_instance.in_waiting)
+                break
+        if time.time() > timeout:
+            logger = Logger()
+            logger.print("Wait UBTN timeout reached", Logger.ERROR)
```

Comment:
I'm not sure but I think that by access your variable using the Logger class imported, you are instantiating a new instance where you just create a new logger a live above. Still digging into this subject by I'm guessing I'm write :)